pkg/flag/options.go

```diff
@@ -173,19 +175,46 @@ func (o *Options) SetOutputWriter(w io.Writer) {
 
 // OutputWriter returns an output writer.
 // If the output file is not specified, it returns os.Stdout.
-func (o *Options) OutputWriter() (io.Writer, func(), error) {
-	if o.outputWriter != nil {
-		return o.outputWriter, func() {}, nil
+func (o *Options) OutputWriter() (io.Writer, func() error, error) {
+	cleanup := func() error { return nil }
+	switch {
+	case o.outputWriter != nil:
+		return o.outputWriter, cleanup, nil
+	case o.Output == "":
+		return os.Stdout, cleanup, nil
+	case strings.HasPrefix(o.Output, "plugin="):
+		return o.outputPluginWriter()
 	}
 
-	if o.Output != "" {
-		f, err := os.Create(o.Output)
-		if err != nil {
-			return nil, nil, xerrors.Errorf("failed to create output file: %w", err)
+	f, err := os.Create(o.Output)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("failed to create output file: %w", err)
+	}
+	return f, f.Close, nil
+}
+
+func (o *Options) outputPluginWriter() (io.Writer, func() error, error) {
+	pluginName := strings.TrimPrefix(o.Output, "plugin=")
+
+	pr, pw := io.Pipe()
+	wait, err := plugin.Start(context.TODO(), pluginName, plugin.RunOptions{
+		Args:  o.OutputPluginArgs,
+		Stdin: pr,
+	})
+	if err != nil {
+		return nil, nil, xerrors.Errorf("plugin start: %w", err)
+	}
+
+	cleanup := func() error {
+		if err := pw.Close(); err != nil {
```

Comment:
nit:
```suggestion
		if err = pw.Close(); err != nil {
```

Reply:
Changed in https://github.com/aquasecurity/trivy/pull/4863/commits/67c684634c1f94ff40d16a5c9b8dcf04d3bf6eec